---
title: "Overlap_MicroglialStates"
author: "Hannah Lucas-Clarke"
editor: source
date: "19/08/25"
engine: knitr
toc: true
toc-depth: 3
toc-expand: true
toc-title: Navigate
format:
  html:
    embed-resources: true
---


```{r}
#| label: Load packages
#| error: false
#| warning: false
#| echo: false
#| output: false 
library("tidyverse")
library("gprofiler2")
library("ComplexHeatmap")
library("org.Hs.eg.db")
library("AnnotationDbi")
library("knitr")
library("kableExtra")
library("gridExtra")
library("colorRamp2")
library("viridis")
library("GeneOverlap")
library("readxl")
library("janitor")
library("ComplexHeatmap")
library("circlize")
library("grid")
library("svglite")
```

Bulk RNA seq from human Braak stage 3/4 

Using DEGs that are Padj<0.05, no LFC cut off. 
Cortical Regions grouped 
Sub cortical Regions grouped 
Collapsed regions 
Universe is all the detected genes after cleaning - # 19,708

```{r}
#| label: Load bulk RNA seq data
#| error: false
#| warning: false
#| echo: false
#| output: false 
universe_bulk <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>% 
  dplyr::select(SYMBOL) %>% unique()

Cingulate_up <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>% 
  filter(sig_reg=="Significantly Upregulated") %>% 
  filter(tissue=="C_CTX") %>% 
  dplyr::select(SYMBOL) %>% unique()

Frontal_Up <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>%
  filter(sig_reg=="Significantly Upregulated") %>% 
  filter(tissue=="F_CTX") %>% 
  dplyr::select(SYMBOL) %>%  unique()

Parietal_Up <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>%
  filter(sig_reg=="Significantly Upregulated") %>% 
  filter(tissue=="P_CTX") %>% 
  dplyr::select(SYMBOL) %>% unique() 

Temporal_Up <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>%
  filter(sig_reg=="Significantly Upregulated") %>% 
  filter(tissue=="T_CTX") %>% 
  dplyr::select(SYMBOL) %>% unique() 

cortical_up <- c(Cingulate_up$SYMBOL,Frontal_Up$SYMBOL,Parietal_Up$SYMBOL,Temporal_Up$SYMBOL) %>% unique()

Cingulate_down <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>% 
  filter(sig_reg=="Significantly Downregulated") %>% 
  filter(tissue=="C_CTX") %>% 
  dplyr::select(SYMBOL) %>% unique()

Frontal_Down <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>%
  filter(sig_reg=="Significantly Downregulated") %>% 
  filter(tissue=="F_CTX") %>% 
  dplyr::select(SYMBOL) %>% unique()

Parietal_Down <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>%
  filter(sig_reg=="Significantly Downregulated") %>% 
  filter(tissue=="P_CTX") %>% 
  dplyr::select(SYMBOL) %>% unique() 

Temporal_Down <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>%
  filter(sig_reg=="Significantly Downregulated") %>% 
  filter(tissue=="T_CTX") %>% 
  dplyr::select(SYMBOL) %>% unique()

cortical_down <- c(Cingulate_down$SYMBOL,Frontal_Down$SYMBOL,Parietal_Down$SYMBOL,Temporal_Down$SYMBOL) %>% unique()

Collapsed_up <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>% 
  filter(sig_reg=="Significantly Upregulated") %>% 
  filter(tissue=="Collapse-All Brain") %>% 
  dplyr::select(SYMBOL) %>% unique()

Collapsed_down <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>% 
  filter(sig_reg=="Significantly Downregulated") %>% 
  filter(tissue=="Collapse-All Brain") %>% 
  dplyr::select(SYMBOL) %>% unique()

caudate_up <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>% 
  filter(sig_reg=="Significantly Upregulated") %>% 
  filter(tissue=="CAU") %>% 
  dplyr::select(SYMBOL) %>% unique()

caudate_down <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>% 
  filter(sig_reg=="Significantly Downregulated") %>% 
  filter(tissue=="CAU") %>% 
  dplyr::select(SYMBOL) %>% unique()

SN_up <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>% 
  filter(sig_reg=="Significantly Upregulated") %>% 
  filter(tissue=="SN") %>% 
  dplyr::select(SYMBOL) %>%  unique()

SN_down <- read.csv("./OurData/Bulk/dge_all_Gene_panel_studies_lookups_added.csv") %>% 
  filter(sig_reg=="Significantly Downregulated") %>% 
  filter(tissue=="SN") %>% 
  dplyr::select(SYMBOL) %>% unique()

subcortical_up <- c(caudate_up$SYMBOL,SN_up$SYMBOL) %>% unique()
subcortical_down <- c(caudate_down$SYMBOL,SN_down$SYMBOL) %>% unique()

bulk_all <- list(Collapsed_up$SYMBOL,Collapsed_down$SYMBOL,cortical_up,cortical_down,subcortical_up,subcortical_down) 
names(bulk_all) <- c("Collapsed_up","Collapsed_down","Cortical_up","Cortical_down","Subcortical_up","Subcortical_down") 

rm(Cingulate_up,Cingulate_down,Frontal_Up,Frontal_Down,Parietal_Up,Parietal_Down,Temporal_Down,Temporal_Up,Collapsed_up,Collapsed_down,caudate_up, caudate_down,SN_up,SN_down,cortical_down,cortical_up,subcortical_down,subcortical_up)
```



**Disease Associated Microglia: DAM (Keren-Shaul 2017)**

Had to convert to human symbols - did this with biomart gui 

```{r}
#| label: DAM
#| error: false
#| warning: false
#| echo: false
#| output: false 
DAM_Up <- read_csv("./MicrogliaPhenotypeLists/DAM_UP_Hs.csv",show_col_types = FALSE) %>% 
  dplyr::select(`Human gene name`) %>% unique() %>% na.omit()
DAM_Down <- read_csv("./MicrogliaPhenotypeLists/DAM_down_Hs.csv",show_col_types = FALSE) %>% 
  dplyr::select(`Human gene name`) %>% unique() %>% na.omit()
```

**Human Alzheimer's Microglia: HAM (Srinivasan 2020)**

```{r}
#| label: HAM
#| error: false
#| warning: false
#| echo: false
#| output: false 
HAM_Up <- read.csv("./MicrogliaPhenotypeLists/HAM_list.csv") %>% filter(Human.AD.Myeloid=="Up") %>% dplyr::select(Gene.Symbol) %>% unique()
HAM_Down <- read.csv("./MicrogliaPhenotypeLists/HAM_list.csv") %>% filter(Human.AD.Myeloid=="Down") %>% dplyr::select(Gene.Symbol) %>% unique()
```


**HuMiCA (Martins-Ferreira 2025) **
Combines data from 19 experiments across different diseases 
pseudo bulked 
disease-inflammatory macrophages - DIM (cluster 2) - AD and LBD 
disease associate microglia - DAM (clusters 1, 3, 5, 6) - AD and LBD 

Only has upregulated genes

```{r}
#| label: Humica
#| error: false
#| warning: false
#| echo: false
#| output: false 
Humica_DAM_AD_Up <- read_csv("./MicrogliaPhenotypeLists/Humica_DAM.csv", show_col_types = FALSE) %>% 
  filter(Group=="AD") %>% filter(log2FoldChange>0) %>% dplyr::select(Gene) %>% unique()
Humica_DAM_LBD_Up <- read_csv("./MicrogliaPhenotypeLists/Humica_DAM.csv",show_col_types = FALSE) %>% 
  filter(Group=="LBD") %>% filter(log2FoldChange>0) %>% dplyr::select(Gene) %>% unique()
Humica_DIM_AD_Up <- read_csv("./MicrogliaPhenotypeLists/Humica_DIM.csv",show_col_types = FALSE) %>% 
  filter(Group=="AD") %>% filter(log2FoldChange>0) %>% dplyr::select(Gene) %>% unique()
Humica_DIM_LBD_Up <- read_csv("./MicrogliaPhenotypeLists/Humica_DIM.csv",show_col_types = FALSE) %>% 
  filter(Group=="LBD") %>% filter(log2FoldChange>0) %>% dplyr::select(Gene) %>% unique()
```

```{r}
#| label: combine datasets
#| error: false
#| warning: false
#| echo: false
#| output: false 
all_external <- list(DAM_Up$`Human gene name`,
                     DAM_Down$`Human gene name`,
                     HAM_Up$Gene.Symbol,
                     HAM_Down$Gene.Symbol,
                     Humica_DAM_AD_Up$Gene,
                     Humica_DAM_LBD_Up$Gene,
                     Humica_DIM_AD_Up$Gene,
                     Humica_DIM_LBD_Up$Gene)
names(all_external) <- c("DAM_Up",
                         "DAM_Down", 
                         "HAM_Up",
                         "HAM_Down",
                         "Humica_DAM_AD_Up",
                         "Humica_DAM_LBD_Up",
                         "Humica_DIM_AD_Up",
                         "Humica_DIM_LBD_Up")

rm(Humica_DAM_AD_Up,Humica_DAM_LBD_Up,Humica_DIM_AD_Up,Humica_DIM_LBD_Up,HAM_Down,HAM_Up, DAM_Down,DAM_Up)
```

```{r}
#| label: Overlap - heatmap
#| error: false
#| warning: false
#| echo: false
#| output: true
#| fig-width: 10
#| fig-height: 10
overlap <- newGOM(bulk_all, all_external, genome.size=nrow(universe_bulk))
pval <- getMatrix(overlap, name="pval")
OR <- getMatrix(overlap, name="odds.ratio")

ngenes <- getMatrix(overlap, name="intersection")
ngenes <- as.data.frame(ngenes)
write.csv(ngenes,file="./bulk_ngenes_overlap.csv",row.names = TRUE, col.names = TRUE)

adj1 <- matrix(p.adjust(as.vector(as.matrix(pval[,1:2])),method="BH"), ncol=2, nrow=6)
adj2 <- matrix(p.adjust(as.vector(as.matrix(pval[,3:4])),method="BH"), ncol=2, nrow=6)
adj3 <- matrix(p.adjust(as.vector(as.matrix(pval[,5:8])),method="BH"), ncol=4, nrow=6)

rownames(adj1) <- rownames(pval)
rownames(adj2) <- rownames(pval)
rownames(adj3) <- rownames(pval)

colnames(adj1) <- colnames(pval)[1:2]
colnames(adj2) <- colnames(pval)[3:4]
colnames(adj3) <- colnames(pval)[5:8]

adj_bulk <- cbind(adj1,adj2,adj3)
rm(adj1,adj2,adj3)

log10_bulk <- -log10(adj_bulk)

myPal <- colorRamp2(seq(0, max(OR), length.out = 100), viridis(100))

# Define row annotation
annot <- data.frame(Group = factor(c(rep("Collapsed Brain", 2), rep("Cortical", 2), rep("Subcortical", 2))),
                    Direction = factor(c("Up", "Down", "Up", "Down", "Up", "Down")))


# Define colors for annotations
annot_colors <- list(Group = c("Collapsed Brain" = "#dea41d",
                               "Cortical" = "#5db6e0",
                               "Subcortical" = "#009b71"),
                     Direction = c("Up" = "#595959", "Down" = "#BFBFBF"))

# Create row annotation object
row_ha_bulk <- rowAnnotation(df = annot, col = annot_colors)

svglite(filename = "bulk_heatmap.svg", width=7, height=6)

heatmap_overlap <- Heatmap(OR,
        name = "Odds Ratio",
        col = myPal,
        cluster_rows = FALSE,
        cluster_columns = FALSE,
        row_title= "Bulk RNASeq PD vs. Control",
        row_title_gp = gpar(fontsize = 10, fontface = "bold"),
        show_column_names = TRUE,
        column_split = c(rep("Keren-Shaul 2017", 2),rep("Srinivasan 2020", 2),rep("Martins-Ferreira 2025", 4)),
        column_names_gp = gpar(fontsize = 10),
        show_row_names = FALSE,
        left_annotation = row_ha_bulk,
        column_title_gp = gpar(fontsize = 10, fontface = "bold"),
        column_title_side = "top",
        column_title_rot =  0,
        cell_fun = function(j, i, x, y, w, h, fill) {
          if (adj_bulk[i, j] < 0.0001) {
            grid.text("****", x, y, gp = gpar(fontface = "bold", fontsize=14))
          } else if (adj_bulk[i, j] < 0.001) {
            grid.text("***", x, y, gp = gpar(fontface = "bold", fontsize=14))
          } else if (adj_bulk[i, j] < 0.01) {
            grid.text("**", x, y, gp = gpar(fontface = "bold", fontsize=14))
          } else if (adj_bulk[i, j] < 0.05) {
            grid.text("*", x, y, gp = gpar(fontface = "bold", fontsize=14))
          }
        })
draw(heatmap_overlap)
dev.off()

write.csv(adj_bulk,"./results_padj_bulk.csv")
write.csv(OR,"./results_OR_bulk.csv")


```